### ISSUE TITLE
How to authentificate client against API with C2E enabled?

### ISSUE THREAD

**Issue by ghost** [marius-wieschollek/passwords#228]
**Current Status**
If I try to authentificate a client against the API and C2E is enabled, the server responds the following:

`{"challenge": { "salts": [ "xxxxxxxxxxxxxxxxx etc.", "xxxxxxxx etc.",  
      "xxxxx etc." ], "type": "PWDv1r1" } }`

It's not clear how to authentificate a client with C2E enabled. The API manual doesn't provide that information.

**Feature Description**
The API manual describes how to authentificate with C2E enabled.

**Additional context**

I started writing an Android client for Passwords in Dart. I've nearly finished the API stuff (the authentification with only SSE enabled is working), but now (in perspective of the upcoming C2E) I'm now working to implement E2E encryption. After reading the manual and studying the source code, however, it is still not entirely clear to me how to authenticate my client with activated E2E encryption.

Could you clarify this a little bit?

**Comment by marius-wieschollek** [comment]
I didn't know that someone was already working on it. I have added the basic information to the [wiki](https://git.mdns.eu/nextcloud/passwords/wikis/Developers/Index) and will add the details for the session, token and account api later this week. 

I have also pushed the latest development status to the javascript [passwords-client](https://git.mdns.eu/nextcloud/passwords-client) package which contains a cleaner and probably easier to understand implementation of the encryption.
